src/utils/markers.ts

```diff
@@ -0,0 +1,88 @@
+import type { SetStateAction, Dispatch } from 'react';
+import MarkerTemplate from '@/components/Map/MarkerTemplate';
+import type { MapAndMarkers } from '@/types/map';
+
+interface MakeCityMarker extends MapAndMarkers {
+  setCityDustInfoMarkers: Dispatch<SetStateAction<kakao.maps.CustomOverlay[]>>;
+}
+
+interface MakeSidoMarker extends MapAndMarkers {
+  allLocation: {
+    location: string;
+    latitude: number;
+    longitude: number;
+  }[];
+}
+
+export const makeCityMarker = ({
+  map,
+  dustInfoList,
+  markers,
+  setCityDustInfoMarkers,
+}: MakeCityMarker) => {
+  if (!map) return;
+
+  const geocoder = new kakao.maps.services.Geocoder();
+
+  dustInfoList.forEach((dustInfoItem) => {
+    geocoder.addressSearch(dustInfoItem.location, (result, status) => {
+      if (status !== kakao.maps.services.Status.OK) return;
+
+      const latitude = Number(result[0].y);
+      const longitude = Number(result[0].x);
```

Comment:
```suggestion
    geocoder.addressSearch(dustInfoItem.location, ([result], status) => {
      if (status !== kakao.maps.services.Status.OK) return;

      const latitude = Number(result.y);
      const longitude = Number(result.x);
```

이렇게 작성하면 `0`이라는 인덱스 하드코딩을 막을 수 있을 것 같아요!

Reply:
정작 필요한 곳에서는 구조 분해 할당을 사용하지 않고 있었군요.
덕분에 배열 구조분해 할당에 대해서 상기하게 되었습니다.
감사합니다 :D